backends/p4tools/modules/testgen/core/small_step/extern_stepper.cpp

```diff
@@ -164,29 +164,15 @@ void ExprStepper::evalInternalExternMethodCall(const IR::MethodCallExpression *c
              auto &nextState = state.clone();
              const auto *prependType = state.resolveType(prependVar->type);
 
-             if (prependType->is<IR::Type_StructLike>()) {
-                 auto prependRef = ToolsVariables::convertReference(prependVar);
-                 // We only support flat assignments, so retrieve all fields from the input
-                 // argument.
-                 const auto flatFields = nextState.getFlatFields(prependRef);
-                 // Iterate through the fields in reverse order.
```

Comment:
Why we don't have to reverse any more?

Reply:
Previously we were iterating over the individual members when prepending. If you started from the beginning and prepend you would invert the order of fields. Now we just prepend the entire struct expression and order is preserved.

Reply:
I understand why the reverse was there. What confuses me is that there is still an iteration over the flattened expression, so I would expect the reverse is still needed.

Reply:
You are right, the reverse iterator should still be there. I was looking at the wrong piece of code...

We are passing a struct expression but once we unroll that struct expression we still need to do that in reverse.